test/utils/clean/clean.go

```diff
@@ -3,39 +3,50 @@ package clean
 import (
 	"context"
 	"fmt"
-	"time"
 
 	"github.com/openshift/ptp-operator/test/utils"
 	"github.com/openshift/ptp-operator/test/utils/client"
+	"github.com/sirupsen/logrus"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func DeleteLabel(label string) error {
+	nodeList, err := client.Client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=", label)})
+	if err != nil {
+		return fmt.Errorf("failed to retrieve grandmaster node list %v", err)
+	}
+	for nodeIndex := range nodeList.Items {
+		delete(nodeList.Items[nodeIndex].Labels, label)
+		_, err = client.Client.CoreV1().Nodes().Update(context.Background(), &nodeList.Items[nodeIndex], metav1.UpdateOptions{})
+		if err != nil {
+			logrus.Errorf("Error updating node, err=%s", err)
+		}
+	}
+	return nil
+}
+
 // All removes any configuration applied by ptp tests.
 func All() error {
 	err := Configs()
 	if err != nil {
 		return err
 	}
 
-	nodeList, err := client.Client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=", utils.PtpGrandmasterNodeLabel)})
+	err = DeleteLabel(utils.PtpGrandmasterNodeLabel)
 	if err != nil {
-		return fmt.Errorf("clean.All: Failed to retrieve grandmaster node list %v", err)
+		return fmt.Errorf("clean.All: fail to delete label: %s, err: %s", utils.PtpGrandmasterNodeLabel, err)
 	}
-	for _, node := range nodeList.Items {
-		delete(node.Labels, utils.PtpGrandmasterNodeLabel)
-		_, err = client.Client.CoreV1().Nodes().Update(context.Background(), &node, metav1.UpdateOptions{})
+	err = DeleteLabel(utils.PtpSlaveNodeLabel)
+	if err != nil {
+		return fmt.Errorf("clean.All: fail to delete label: %s, err: %s", utils.PtpSlaveNodeLabel, err)
 	}
-
-	nodeList, err = client.Client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=", utils.PtpSlaveNodeLabel)})
+	err = DeleteLabel(utils.PtpBCMasterNodeLabel)
```

Comment:
Isn't the PTP Master and GM  labels mean same ?

Reply:
This label is to identify the boundary clock, the other new label is for boundary clock slave